src/result.rs

```diff
@@ -163,235 +158,107 @@ impl RowWrapper {
-            None => Err(Self::generic_error("inet")),
-        }
-    }
-
-    #[napi]
-    pub fn get_list(&self) -> napi::Result<Vec<CqlValueWrapper>> {
-        match self.inner.as_list() {
-            Some(r) => Ok(r
-                .iter()
-                .map(|f| CqlValueWrapper { inner: f.clone() })
-                .collect()),
-            None => Err(Self::generic_error("list")),
-        }
-    }
-
-    #[napi]
-    /// Return array of "tuples" keeping (key, value) for each element in the map
-    /// Currently NAPI doesn't implement ToNapiValue for tuples,
-    /// but the returned typed here will be the same as if we returned tuple.
-    pub fn get_map(&self) -> napi::Result<Vec<Vec<CqlValueWrapper>>> {
-        match self.inner.as_map() {
-            Some(r) => Ok(r
-                .iter()
-                .map(|f| {
-                    vec![
-                        CqlValueWrapper { inner: f.0.clone() },
-                        CqlValueWrapper { inner: f.1.clone() },
-                    ]
-                })
-                .collect()),
-            None => Err(Self::generic_error("map")),
-        }
-    }
-
-    #[napi]
-    pub fn get_set(&self) -> napi::Result<Vec<CqlValueWrapper>> {
-        match self.inner.as_set() {
-            Some(r) => Ok(r
-                .iter()
-                .map(|f| CqlValueWrapper { inner: f.clone() })
-                .collect()),
-            None => Err(Self::generic_error("set")),
-        }
-    }
-
-    #[napi]
-    pub fn get_small_int(&self) -> napi::Result<i16> {
-        match self.inner.as_smallint() {
-            Some(r) => Ok(r),
-            None => Err(Self::generic_error("small_int")),
-        }
-    }
-
-    #[napi]
-    pub fn get_tiny_int(&self) -> napi::Result<i8> {
-        match self.inner.as_tinyint() {
-            Some(r) => Ok(r),
-            None => Err(Self::generic_error("tiny_int")),
-        }
-    }
-    #[napi]
-    pub fn get_local_time(&self) -> napi::Result<LocalTimeWrapper> {
-        match self.inner.as_cql_time() {
-            Some(r) => Ok(LocalTimeWrapper::from_cql_time(r)),
-            None => Err(Self::generic_error("local_time")),
-        }
-    }
-
-    #[napi]
-    pub fn get_time_uuid(&self) -> napi::Result<TimeUuidWrapper> {
-        match self.inner.as_timeuuid() {
-            Some(r) => Ok(TimeUuidWrapper::from_cql_time_uuid(r)),
-            None => Err(Self::generic_error("time_uuid")),
-        }
-    }
+/// Returns tuple of elements:
+/// (typ, value)
+/// for a given CQL Value.
+///
+/// Intended to be used when there is ambiguity how the given value
+/// should be provided, based on it's type.
+///
+/// For example CQL::BigInt and CqlValue::Timestamp are returned from rust layer
+/// as BigInt, but are provided as BigInt and Date.
+unsafe fn add_type_to_napi_obj(
+    env: napi::sys::napi_env,
+    value: napi::Result<napi::sys::napi_value>,
+    typ: CqlType,
+) -> napi::Result<napi::sys::napi_value> {
```

Comment:
:wrench: If you are writing any `unsafe` function, you must provide a docstring that lists assumptions which needs to be ensured for safety.

Similarly, if you are writing any `unsafe` block, you must provide a comment which proves that all the necessary assumptions are ensured.